x/fungible/types/message_update_gateway_contract.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+const TypeMsgUpdateGatewayContract = "update_gateway_contract"
+
+var _ sdk.Msg = &MsgUpdateGatewayContract{}
+
+func NewMsgUpdateGatewayContract(creator string, gatewayContractAddr string) *MsgUpdateGatewayContract {
+	return &MsgUpdateGatewayContract{
+		Creator:                   creator,
+		NewGatewayContractAddress: gatewayContractAddr,
+	}
+}
+
+func (msg *MsgUpdateGatewayContract) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgUpdateGatewayContract) Type() string {
+	return TypeMsgUpdateGatewayContract
+}
+
+func (msg *MsgUpdateGatewayContract) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
```

Comment:
Let's not use panics here. Can observer&signer technically send a message with some artbitrary creator e.g. `abc`?

Reply:
I think it makes sense to remove the panic, we used this pattern for all message so far, so I think it's better to create an issue first, do some assessment of the change and change for all messages at once

Reply:
https://github.com/zeta-chain/node/issues/2592